plugin/trino-postgresql/src/test/java/io/trino/plugin/postgresql/TestPostgreSqlCastPushdown.java

```diff
@@ -0,0 +1,432 @@
+                new CastTestCase("c_decimal_19_2", "tinyint", "c_smallint"),
+                new CastTestCase("c_decimal_30_2", "tinyint", "c_smallint"),
+                new CastTestCase("c_decimal_negative", "tinyint", "c_smallint"),
+
+                new CastTestCase("c_real", "tinyint", "c_smallint"),
+                new CastTestCase("c_float4", "tinyint", "c_smallint"),
+                new CastTestCase("c_double_precision", "tinyint", "c_smallint"),
+                new CastTestCase("c_float", "tinyint", "c_smallint"),
+                new CastTestCase("c_float8", "tinyint", "c_smallint"),
+                new CastTestCase("c_varchar_numeric", "tinyint", "c_smallint"),
+                new CastTestCase("c_text_numeric", "tinyint", "c_smallint"),
+                new CastTestCase("c_varchar_numeric_sign", "tinyint", "c_smallint"),
+
+                new CastTestCase("c_real", "smallint", "c_smallint"),
+                new CastTestCase("c_float4", "smallint", "c_smallint"),
+                new CastTestCase("c_double_precision", "smallint", "c_smallint"),
+                new CastTestCase("c_float", "smallint", "c_smallint"),
+                new CastTestCase("c_float8", "smallint", "c_smallint"),
+                new CastTestCase("c_varchar_numeric", "smallint", "c_smallint"),
+                new CastTestCase("c_text_numeric", "smallint", "c_smallint"),
+                new CastTestCase("c_varchar_numeric_sign", "smallint", "c_smallint"),
+
+                new CastTestCase("c_real", "integer", "c_integer"),
+                new CastTestCase("c_float4", "integer", "c_integer"),
+                new CastTestCase("c_double_precision", "integer", "c_integer"),
+                new CastTestCase("c_float", "integer", "c_integer"),
+                new CastTestCase("c_float8", "integer", "c_integer"),
+                new CastTestCase("c_varchar_numeric", "integer", "c_integer"),
+                new CastTestCase("c_text_numeric", "integer", "c_integer"),
+                new CastTestCase("c_varchar_numeric_sign", "integer", "c_integer"),
+
+                new CastTestCase("c_real", "bigint", "c_bigint"),
+                new CastTestCase("c_float4", "bigint", "c_bigint"),
+                new CastTestCase("c_double_precision", "bigint", "c_bigint"),
+                new CastTestCase("c_float", "bigint", "c_bigint"),
+                new CastTestCase("c_float8", "bigint", "c_bigint"),
+                new CastTestCase("c_varchar_numeric", "bigint", "c_bigint"),
+                new CastTestCase("c_text_numeric", "bigint", "c_bigint"),
+                new CastTestCase("c_varchar_numeric_sign", "bigint", "c_bigint"),
+
+                new CastTestCase("c_smallint", "boolean", "c_boolean"),
+                new CastTestCase("c_real", "double", "c_double_precision"),
+                new CastTestCase("c_double_precision", "real", "c_real"),
+                new CastTestCase("c_double_precision", "decimal(10,2)", "c_decimal_10_2"));
```

Comment:
You might want to put it a few negative test cases for other unsupported types as target (varchar, char, etc) - although this can be done in the followup  as well along with more types

Reply:
Added smallint, integer, double, numeric types negative tests.
The unsupported type target with char/varchar I would like to leave it to the char/varchar support in the follow ups.